MAUI/DataGrid/Column-types.md

```diff
@@ -0,0 +1,916 @@
+        </DataTemplate>
+    </ResourceDictionary>
+</ContentPage.Resources>
+
+<sfgrid:DataGridTemplateColumn MappingName="Freight" >
+    <sfgrid:DataGridTemplateColumn.CellTemplate>
+        <local:FreightTemplateSelector High="{StaticResource high}"
+                                       Average="{StaticResource average}"
+                                       Low="{StaticResource low}"/>
+    </sfgrid:DataGridTemplateColumn.CellTemplate>
+</sfgrid:DataGridTemplateColumn>
+{% endhighlight %}
+
+{% highlight c# %}
+// FreightTemplateSelector implementation
+public class FreightTemplateSelector : DataTemplateSelector
+{
+    public DataTemplate Low { get; set; }
+
+    public DataTemplate Average { get; set; }
+
+    public DataTemplate High { get; set; }
+
+    protected override DataTemplate OnSelectTemplate(object item, BindableObject container)
+    {
+        var value = double.Parse((item as OrderInfo).Freight);
+        if (value > 750)
+            return High;
+        else if (value > 500)
+            return Average;
+        else
+            return Low;
+    }
+}
+{% endhighlight %}
+
+![DateColumn](Images/column-types/template-column.PNG)
+
+### Getting row index of a row in GridTemplateColumn
+
+The SfDataGrid provides various resolving methods to resolve the row index of grid rows based on certain criteria. The actual row index of a row can be resolved by using the `ResolveToRowIndex(recordRowIndex)` method. 
+
+The row index of a grid row can be obtained in GridTemplateColumn by retrieving the record index of the row using the bound data from its `BindingContext` and by resolving the recordRowIndex using the `SfDataGrid.ResolveToRowIndex(recordRowIndex)` method.
+
+{% tabs %}
+{% highlight xaml %}
+// MainPage.Xaml
+
+<sfgrid:DataGridTemplateColumn HeaderText="ShipCity" MappingName="ShipCity">
+  <sfgrid:DataGridTemplateColumn.CellTemplate>
+    <DataTemplate>
+      <Button Clicked="button_Clicked" WidthRequest="120" Text="{Binding ShipCity}"/>
+    </DataTemplate>
+  </sfgrid:DataGridTemplateColumn.CellTemplate>
+</sfgrid:DataGridTemplateColumn>
+
+{% endhighlight %}
+
+{% highlight c# %}
+// MainPage.cs 
+public partial class MainPage : ContentPage
+{
+     public MainPage()
+     {
+         InitializeComponent();
+     }
+     private void button_Clicked(object sender, EventArgs e)
+     {
+          var button = sender as Button;
+          var record = button.BindingContext as OrderInfo;
+          var recordRowIndex = viewModel.OrderInfoCollection.IndexOf(record);
+          var rowIndex = sfGrid.ResolveToRowIndex(recordRowIndex);
+     }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## DataGridDateColumn
+
+The `SfDataGrid.DataGridDateColumn` inherits all the properties of the `SfDataGrid.DataGridColumn`. It displays the date information as the content of a column. To create the `SfDataGrid.DataGridDateColumn`, the property corresponding to the column in the underlying collection must be of type DateTime. 
```

Comment:
The `SfDataGrid.DataGridDateColumn` inherits all the properties of the `SfDataGrid.DataGridColumn`. It displays the date information as the content of a column. To create the `SfDataGrid.DataGridDateColumn`, the property corresponding to the column in the underlying collection must be of the type DateTime.